### ISSUE TITLE
Does react-styled-select support Async options like react-select ?

### ISSUE THREAD

**Issue by yasserhennawi** [agutoli/react-styled-select#13]
I was wondering if (by any mean) react-styled-select is currently supporting Async options, as react-select currently do with <Select.Sync loadOptions={foo} /> but the exact same code doesn't work here, Or let me rephrase, Apparently it neither of <Select.Sync /> nor loadOptions/asyncOptions props.

**Comment by agutoli** [comment]
@yasserhennawi Actually not. This should be implemented, I will try to implement it soon following the `react-select` api/signature.

**Comment by yasserhennawi** [comment]
Cool waiting for that eagerly! keep up the good work @agutoli  :+1:

**Comment by agutoli** [comment]
@yasserhennawi Good? I did a "basic" implementation for `Select.Async`. 

you can try the version `2.1.0-alpha`

the API/signature it`s the soma from `react-select` as well

`var getOptions = function(input, callback) {
  setTimeout(function() {
    callback(null, {
      options: [
        { value: 'one', label: 'One' },
        { value: 'two', label: 'Two' }
      ],
      // CAREFUL! Only set this to true when there are no more options,
      // or more specific queries will not be sent to the server.
      complete: true
    });
  }, 500);
};

<Select.Async
    name="form-field-name"
    loadOptions={getOptions}
/>`

**Comment by yasserhennawi** [comment]
Hey @agutoli apologies for the late response, Working perfect! Thanks alot 👍
